wdl2cwl/main_oop.py

```diff
@@ -66,76 +66,72 @@ def load_wdl_task(self, obj: WDL.Tree.Task) -> str:
         """Load task and convert to CWL."""
         cwl_inputs = self.get_cwl_inputs(obj.inputs)
         cwl_outputs = self.get_cwl_outputs(obj.outputs)
-        runtime_docker = obj.runtime["docker"]
-        if not isinstance(runtime_docker, (WDL.Expr.Get, WDL.Expr.String)):
-            raise Exception(
-                f"Unsupported docker runtime type: {type(runtime_docker)}: {runtime_docker}"
-            )
-        docker_requirement = self.get_cwl_docker_requirements(runtime_docker)
-        cwl_command_str = self.get_cwl_command_requirements(obj.command.parts)
-        base_command = ["bash", "example.sh"]
-        requirements: List[cwl.ProcessRequirement] = [
-            docker_requirement,
-            cwl_command_str,
-        ]
-        requirements.append(cwl.InlineJavascriptRequirement())
-        requirements.append(cwl.NetworkAccess(networkAccess=True))
-        cpu_requirement = (
-            self.get_cpu_requirement(obj.runtime["cpu"])
-            if "cpu" in obj.runtime
-            else None
-        )
-        memory_requirement = (
-            self.get_memory_requirement(obj.runtime["memory"])  # type: ignore
-            if "memory" in obj.runtime
-            else None
-        )
-        outdir_requirement = (
-            self.get_outdir_requirement(obj.runtime["disks"])  # type: ignore
-            if "disks" in obj.runtime
-            else 1024
-        )
-        requirements.append(
-            cwl.ResourceRequirement(
-                coresMin=cpu_requirement,
-                ramMin=memory_requirement,
-                outdirMin=outdir_requirement,
-            )
-        )
+        requirements = self.get_cwl_requirements(obj.command, obj.runtime)
 
-        cat_tool = cwl.CommandLineTool(
+        cmd_tool = cwl.CommandLineTool(
             id=obj.name,
             inputs=cwl_inputs,
-            requirements=requirements,
             outputs=cwl_outputs,
+            requirements=requirements,
             cwlVersion="v1.2",
-            baseCommand=base_command,
+            baseCommand=["bash", "example.sh"],
         )
 
         yaml = YAML()
         yaml.default_flow_style = False
         yaml.indent = 4
         yaml.block_seq_indent = 2
-        result_stream = StringIO()
-        cwl_result = cat_tool.save()
+        cwl_result = cmd_tool.save()
         scalarstring.walk_tree(cwl_result)
+        result_stream = StringIO()
         yaml.dump(cwl_result, result_stream)
         yaml.dump(cwl_result, sys.stdout)
 
         return result_stream.getvalue()
 
-    def get_outdire_requirement(
+    def get_cwl_requirements(
+        self, command: WDL.Expr.String, runtime: WDL.Dict[str, WDL.Expr.Base]
```

Comment:
```suggestion
        self, command: WDL.Expr.String, runtime: Dict[str, WDL.Expr.Base]
```

and add `Dict` to the `from typing import List, Union, Optional, Callable, cast, Any, Set` line at the top :-)